File: 2. Medium/Python/Add Two Numbers.py
def addTwoNumbers(l1, l2):
        num1 , num2 = '' , ''
        
        while l1: # iterate first list
            num1 = str(l1.val) + num1
            l1 = l1.next
            
        while l2: # iterate second list
            num2 = str(l2.val) + num2
            l2 = l2.next

        summ = str(int(num1) + int(num2))
        temp = ListNode(int(summ[0]))
        
        for i in summ[1:]: # create node for every digit
            newNode = ListNode(int(i),temp)
            temp    = newNode

        return temp

File: 1. Easy/Java/Add Two Numbers.java
class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        ListNode head = null;
        ListNode temp = null;
        int sum;
        boolean ten = false;
        
        //add the individual digits (same position) together.
        while (l1 != null && l2 != null) {
            sum = l1.val + l2.val;
            
            if (ten) {
                sum++;
                ten = false;
            }
            
            if (sum >= 10) {
                sum = sum % 10;
                ten = true;
            }
            
            if (head == null) {
                head = new ListNode(sum);
                temp = head;
            } else {
                temp.next = new ListNode(sum);
                temp = temp.next;
            }
            
            l1 = l1.next;
            l2 = l2.next;
        }
        
        //For num1 and num2 with different orders (e.g. 3 and 86)
        ListNode excess = null;
        if (l1 != null) {
            excess = l1;
        } else if (l2 != null) {
            excess = l2;
        }
        
        while (excess != null) {
            if (ten) {
                excess.val++;
                ten = false;
            }
            
            if (excess.val >= 10) {
                ten = true;
                excess.val = excess.val % 10;
            }
            
            temp.next = new ListNode(excess.val);
            temp = temp.next;
            excess = excess.next;
        }
        
        if (ten) {
            temp.next = new ListNode(1);
        }
        
        
        return head;
    } 
}